src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferenceCache.cs

```diff
@@ -29,6 +29,7 @@ public static FindReferenceCache GetCache(SemanticModel model)
         private readonly SemanticModel _semanticModel;
 
         private readonly ConcurrentDictionary<SyntaxNode, SymbolInfo> _symbolInfoCache = new();
+        private readonly ConcurrentDictionary<SyntaxToken, PreprocessingSymbolInfo> _preprocessingSymbolInfoCache = new();
```

Comment:
Why do we need a new map here? Why doesn't the map right above this work?

Reply:
It failed to be useful as `PreprocessingSymbolInfo` is a struct and seems to have no direct relation to `SymbolInfo`. However, `PreprocessingSymbolInfo` contains redundant information for the reference finder, which is the state of whether the pp symbol is defined or not. It seems like a very primitive construction of a `SymbolInfo` instance could also work.

Reply:
so let me rephrase.  i dn't see the need for a cache at all here.  the caches are useful for the cases when lots of other finders are going to be concurrently querying for semantic info about the same nodes.  That won't be the case for PP symbols. No other finders will run, and you'll only ever do a single pass on the file. so no need to cache at all :)

Reply:
Got it, I'm removing the caching